autogen/agentchat/groupchat.py

```diff
@@ -374,7 +374,7 @@ def run_chat(
                 break
             try:
                 # select the next speaker
-                speaker = groupchat.select_speaker(speaker, self)
+                speaker = groupchat.select_speaker(speaker, speaker)
```

Comment:
The first argument is last_speaker. The second argument is the selector. The original code is correct.

Reply:
You're so right, I have things backwards, I'm so sorry.  Thanks.  I apologize for the waste of time.  I was reading things incorrectly.

Reply:
Still we very much appreciate your PR! if you have future PRs we are happy to take a look.